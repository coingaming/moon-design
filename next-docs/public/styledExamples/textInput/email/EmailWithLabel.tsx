import React from 'react';
import { TextInput } from '@heathmont/moon-core';

const Example = () => (
  <div className="flex flex-col lg:flex-row justify-around items-end w-full gap-2">
    <TextInput
      type="email"
      inputSize="medium"
      label="medium"
      placeholder="e.g. jake.weary@sportsbet.io"
    />
    <TextInput
      type="email"
      inputSize="large"
      label="large"
      placeholder="e.g. jake.weary@sportsbet.io"
    />
    <TextInput
      type="email"
      inputSize="xlarge"
      label="xlarge"
      placeholder="e.g. jake.weary@sportsbet.io"
    />
  </div>
);

export default Example;
