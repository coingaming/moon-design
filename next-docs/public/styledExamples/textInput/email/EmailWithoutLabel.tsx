import React from 'react';
import { TextInput } from '@heathmont/moon-core';

const Example = () => (
  <div className="flex flex-col lg:flex-row justify-around items-end w-full gap-2">
    <TextInput type="email" inputSize="medium" placeholder="medium" />
    <TextInput type="email" inputSize="large" placeholder="large" />
    <TextInput type="email" inputSize="xlarge" placeholder="xlarge" />
  </div>
);

export default Example;
