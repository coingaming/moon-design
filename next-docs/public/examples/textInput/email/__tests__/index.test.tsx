/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { moonDesignLight, ThemeProvider } from '@heathmont/moon-themes';

import EmailWithLabel from '../EmailWithLabel';
import EmailWithoutLabel from '../EmailWithoutLabel';
import EmailWithHint from '../EmailWithHint';
import EmailDisabled from '../EmailDisabled';
import EmailError from '../EmailError';
import EmailBgColor from '../EmailBgColor';
import EmailReadonly from '../EmailReadonly';

const renderWithTheme = (component: JSX.Element) => (
  <ThemeProvider theme={moonDesignLight}>{component}</ThemeProvider>
);

describe('TextInput: type Email', () => {
  it('renders with label', () => {
    const testRenderer = renderer.create(renderWithTheme(<EmailWithLabel />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders without label', () => {
    const testRenderer = renderer.create(
      renderWithTheme(<EmailWithoutLabel />)
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders with hint', () => {
    const testRenderer = renderer.create(renderWithTheme(<EmailWithHint />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders disabled', () => {
    const testRenderer = renderer.create(renderWithTheme(<EmailDisabled />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders error', () => {
    const testRenderer = renderer.create(renderWithTheme(<EmailError />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('supports background colour', () => {
    const testRenderer = renderer.create(renderWithTheme(<EmailBgColor />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders readOnly', () => {
    const testRenderer = renderer.create(renderWithTheme(<EmailReadonly />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
