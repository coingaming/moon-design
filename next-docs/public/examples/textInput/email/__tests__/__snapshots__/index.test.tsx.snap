// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput: type Email renders disabled 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 gBLEES"
      disabled={true}
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 jqAoOL"
        disabled={true}
        placeholder=" "
        type="email"
        value="jake.weary@sportsbet.io"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 gBLEES"
      disabled={true}
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 yCWvw"
        disabled={true}
        placeholder=" "
        type="email"
        value="jake.weary@sportsbet.io"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 gBLEES"
      disabled={true}
    >
      <div
        className="Inner-sc-1soq84i-0 jvXSNE"
      >
        <input
          className="Input-qkdgvx-0 bTDZWF"
          disabled={true}
          placeholder=" "
          type="email"
          value="jake.weary@sportsbet.io"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Email renders error 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 kwpJwN"
        placeholder="e.g. jake.weary@sportsbet.io"
        type="email"
      />
      <p
        className="HintText-sc-1e9hzxr-0 heGnbI"
        role="alert"
      >
        Error message
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 dSfsRu"
        placeholder="e.g. jake.weary@sportsbet.io"
        type="email"
      />
      <p
        className="HintText-sc-1e9hzxr-0 heGnbI"
        role="alert"
      >
        Error message
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 jvXSNE"
      >
        <input
          className="Input-qkdgvx-0 gKjnuf"
          placeholder="e.g. jake.weary@sportsbet.io"
          type="email"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 heGnbI"
        role="alert"
      >
        Error message
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Email renders readOnly 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 jqAoOL"
        placeholder=" "
        readOnly={true}
        type="email"
        value="jake.weary@sportsbet.io"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 yCWvw"
        placeholder=" "
        readOnly={true}
        type="email"
        value="jake.weary@sportsbet.io"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 jvXSNE"
      >
        <input
          className="Input-qkdgvx-0 bTDZWF"
          placeholder=" "
          readOnly={true}
          type="email"
          value="jake.weary@sportsbet.io"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Email renders with hint 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 jqAoOL"
        placeholder="e.g. jake.weary@sportsbet.io"
        type="email"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 yCWvw"
        placeholder="e.g. jake.weary@sportsbet.io"
        type="email"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 jvXSNE"
      >
        <input
          className="Input-qkdgvx-0 bTDZWF"
          placeholder="e.g. jake.weary@sportsbet.io"
          type="email"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Email renders with label 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 jqAoOL"
        placeholder="e.g. jake.weary@sportsbet.io"
        type="email"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 yCWvw"
        placeholder="e.g. jake.weary@sportsbet.io"
        type="email"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 jvXSNE"
      >
        <input
          className="Input-qkdgvx-0 bTDZWF"
          placeholder="e.g. jake.weary@sportsbet.io"
          type="email"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Email renders without label 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <input
        className="Input-qkdgvx-0 jqAoOL"
        placeholder="medium"
        type="email"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <input
        className="Input-qkdgvx-0 yCWvw"
        placeholder="large"
        type="email"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 jvXSNE"
      >
        <input
          className="Input-qkdgvx-0 kwsobc"
          placeholder="xlarge"
          type="email"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Email supports background colour 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 cgHTJu"
        placeholder="e.g. jake.weary@sportsbet.io"
        type="email"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 fGPqtd"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 hAmOHl"
        placeholder="e.g. jake.weary@sportsbet.io"
        type="email"
      />
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 hYWLpt"
      >
        <input
          className="Input-qkdgvx-0 bTDZWF"
          placeholder="e.g. jake.weary@sportsbet.io"
          type="email"
        />
        <label
          className="LabelInner-vrrjt0-0 deQKxJ"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 dfhEvA"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;
